src/main/java/com/neonlab/product/apis/DeleteProductApi.java

```diff
@@ -0,0 +1,33 @@
+package com.neonlab.product.apis;
+import com.neonlab.common.annotations.Loggable;
+import com.neonlab.common.dto.ApiOutput;
+import com.neonlab.common.expectations.InvalidInputException;
+import com.neonlab.common.utilities.StringUtil;
+import com.neonlab.product.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+
+@Service
+@Loggable
+public class DeleteProductApi {
+
+    @Autowired
+    private ProductService productService;
+
+    public ApiOutput<?>deleteProductApi(String code){
+        try {
+            validate(code);
+            return productService.deleteProductApi(code);
+        }catch (InvalidInputException e){
+            return new ApiOutput<>(HttpStatus.BAD_REQUEST.value(), e.getMessage(),null);
```

Comment:
You are using the wrong constructor for this scenario. Please look into the ApiOut class, I have created another constructor for these scenario. Please update.

cc- @sjha24